frontend/app/routes/$lang+/_protected+/personal-information+/mailing-address+/edit.tsx

```diff
@@ -179,95 +217,63 @@ export default function PersonalInformationMailingAddressEdit() {
     })
     .sort((region1, region2) => region1.children.localeCompare(region2.children));
 
-  /**
-   * Gets an error message based on the provided internationalization (i18n) key.
-   *
-   * @param errorI18nKey - The i18n key for the error message.
-   * @returns The corresponding error message, or undefined if no key is provided.
-   */
-  function getErrorMessage(errorI18nKey?: string): string | undefined {
-    if (!errorI18nKey) return undefined;
-
-    /**
-     * The 'as any' is employed to circumvent typechecking, as the type of
-     * 'errorI18nKey' is a string, and the string literal cannot undergo validation.
-     */
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    return t(`personal-information:mailing-address.edit.error-message.${errorI18nKey}` as any);
-  }
-
-  const errorMessages = {
-    streetName: getErrorMessage(actionData?.errors.fieldErrors.streetName?.[0]),
-    cityName: getErrorMessage(actionData?.errors.fieldErrors.cityName?.[0]),
-    provinceTerritoryStateId: getErrorMessage(actionData?.errors.fieldErrors.provinceTerritoryStateId?.[0]),
-    postalCode: getErrorMessage(actionData?.errors.fieldErrors.postalCode?.[0]),
-    countryId: getErrorMessage(actionData?.errors.fieldErrors.countryId?.[0]),
-  };
-
-  const errorSummaryItems = createErrorSummaryItems(errorMessages);
-
-  useEffect(() => {
-    if (actionData?.formData && hasErrors(actionData.formData)) {
-      scrollAndFocusToErrorSummary(errorSummaryId);
-    }
-  }, [actionData]);
-
   return (
     <>
       {errorSummaryItems.length > 0 && <ErrorSummary id={errorSummaryId} errors={errorSummaryItems} />}
-      <Form className="max-w-prose" method="post" noValidate>
+      <fetcher.Form className="max-w-prose" method="post" noValidate>
         <input type="hidden" name="_csrf" value={csrfToken} />
         <div className="my-6">
           <div className="space-y-6">
             <InputField
-              id="streetName"
+              id="street-name"
               className="w-full"
               label={t('personal-information:mailing-address.edit.field.address')}
               helpMessagePrimary={t('personal-information:mailing-address.edit.field.address-note')}
               name="streetName"
               required
-              defaultValue={defaultValues.streetName}
-              errorMessage={errorMessages.streetName}
+              defaultValue={addressInfo.streetName}
+              errorMessage={errorMessages['street-name']}
```

Comment:
Since the validator limits this field to 30 characters, we should add `maxLength={30}` property to this component.

The same comment applies for other input fields where we limit the number of characters.

Reply:
using buildin error message of min(30) as /apply form